android/upload_symbols.gradle

```diff
@@ -0,0 +1,256 @@
+import java.util.zip.ZipOutputStream
+import java.util.zip.ZipEntry
+
+// Custom Exception for undefined variable
+public class VariableNotFoundException extends Exception {
+    public VariableNotFoundException(String errorMessage) {
+        super(errorMessage);
+    }
+}
+// Constants
+final INSTABUG_APP_TOKEN = 'INSTABUG_APP_TOKEN'
+final INSTABUG_API_KEY = 'INSTABUG_API_KEY'
+final UPLOAD_URL = 'https://api.instabug.com/api/web/public/flutter-symbol-files/android'
+final INSTABUG_AUTO_UPLOAD_ENABLE = 'INSTABUG_AUTO_UPLOAD_ENABLE'
+ext.boundary = 'abcd' + Long.toString(System.currentTimeMillis()) * 2 + 'dcba'
+ext.lineEnd = '\r\n'
+
+/**
+ * task to upload symbols to Instabug endpoint
+ */
+task upload_symbols_task {
+    doLast {
+        try {
+            // If obfuscate flag not provided, exit the task
+            if (project['dart-obfuscation'] != 'true') {
+                println "Instabug: skipping debug symbols upload"
+                return
+            }
+            // Load local properties
+            load_local_properties()
+
+            // If auto upload is disabled, exit the task
+            final uploadFlag = get_variable(INSTABUG_AUTO_UPLOAD_ENABLE)
+            println "$INSTABUG_AUTO_UPLOAD_ENABLE: $uploadFlag"
+            if (!uploadFlag?.toBoolean()) {
+                println "Instabug: skipping debug symbols upload"
+                return
+            }
+
+            // Get upload configs
+            final applicationToken = get_variable(INSTABUG_APP_TOKEN);
+            final apiKey = get_variable(INSTABUG_API_KEY);
+            final symbolsPath = get_symbols_path()
+
+            // Prepare zip file with symbols
+            final zipFile = prepare_zip_file(symbolsPath)
+
+            upload_symbols(zipFile, applicationToken, apiKey, UPLOAD_URL)
+        } catch (VariableNotFoundException e) {
+            println "Instabug: uploading symbols: ${e.message}"
+        } catch (Exception e) {
+            println "Instabug err: uploading symbols: ${e.message}"
+        }
+    }
+}
+/**
+ * load local properties into a global variable localProperties
+ */
+def load_local_properties() {
+    // Get local properties
+    final properties = new Properties()
+    final propertiesFile = project.rootProject.file('local.properties')
+    if (propertiesFile.exists()) {
+        properties.load(propertiesFile.newDataInputStream())
+    }
+    ext.localProperties = properties
```

Comment:
I think returning the `properties` instead of assigning it to the global `ext` would be a better approach; to avoid any unwanted changes or overwrites.